Service/Cart.php

```diff
@@ -105,8 +111,12 @@ public function __construct(
         \Magento\CatalogInventory\Api\StockStateInterface $stockState,
         \Magento\CatalogInventory\Model\Quote\Item\QuantityValidator\QuoteItemQtyList $quoteItemQtyList,
         \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig,
-        \Magento\InventorySalesApi\Api\StockResolverInterface $stockResolver,
-        \Magento\InventorySalesApi\Model\GetStockItemDataInterface $stockItemData
+        \MageSuite\FreeGift\Model\Factory\StockResolverFactory $stockResolver,
+        \MageSuite\FreeGift\Model\Factory\GetStockItemDataFactory $getStockItemDatafact,
+        \MageSuite\FreeGift\Model\Factory\SalesChannelFactory $saleschannel,
```

Comment:
@zeel1012 keep the naming consistent. Also for $getStockItemDatafact, use camel case and write it complete like $getStockItemDataFactory